### DISCUSSION TITLE
Embedding a wasm interpreter

### DISCUSSION THREAD

**Discussion by mocenigo** [RustPython/RustPython#5229] [General]
I understand that the intepreter is used if RustPython is embedded in an application, like an editor. Would it be possible to use also, say, wasmtime instead? So the editor would be scripted by RustPython but the code would be compiled to WASM (and maybe from that to native) rather than interpreted?

**Comment by youknowone** [comment]
If you mean running python scripts using RustPython in wasm, it is possible. Please try wasm or wasi target.

**Reply by mocenigo** [reply]
So, for instance, a text editor that is scripted with RustPython would be able to use RustPython as an interpreter AND also compile the program to WASM with WASI and have the same functionality (but potentially better speed). Cool.

**Reply by youknowone** [reply]
rustpython doesnt compile python code to wasm, if you meant that. It will not improve performance.

**Comment by mocenigo** [comment]
> rustpython doesnt compile python code to wasm, if you meant that. It will not improve performance.

The project description states “ RustPython is a Python interpreter written in Rust. RustPython can be embedded into Rust programs to use Python as a scripting language for your application, **or it can be compiled to WebAssembly in order to run Python in the browser.** ”

So that’s where my confusion arose. It is not immediately clear.
